src/unlimited.cpp

```diff
@@ -637,12 +640,12 @@ void UnlimitedSetup(void)
     recvAmt.init("net/recv/total");
     recvAmt.init("net/send/total");
     std::vector<std::string> msgTypes = getAllNetMessageTypes();
-    
+   
     for (std::vector<std::string>::const_iterator i=msgTypes.begin(); i!=msgTypes.end();++i)
-      {
-	new CStatHistory<uint64_t >("net/recv/msg/" +  *i);  // This "leaks" in the sense that it is never freed, but is intended to last the duration of the program.
-	new CStatHistory<uint64_t >("net/send/msg/" +  *i);  // This "leaks" in the sense that it is never freed, but is intended to last the duration of the program.
-      }
+    {
+        mallocedStats.push_front(new CStatHistory<uint64_t >("net/recv/msg/" +  *i));
```

Comment:
It's worth mentioning you can still shove this into a `unique_ptr` even if it is meant to last the program duration. In this case cleanup happens on shutdown, just either make it `static` or place it in global scope.